pkg/daemon/update.go

```diff
@@ -2180,7 +2180,7 @@ func (dn *CoreOSDaemon) applyLayeredOSChanges(mcDiff machineConfigDiff, oldConfi
 
 	var osExtensionsContentDir string
 	var err error
-	if mcDiff.osUpdate || mcDiff.extensions || mcDiff.kernelType {
+	if newConfig.Spec.BaseOSExtensionsContainerImage != "" && (mcDiff.osUpdate || mcDiff.extensions || mcDiff.kernelType) {
```

Comment:
Instead of doing this, we should ignore this section for OKD with conditional if. Currently checking for empty means that we may silently ignore extensions if somehow missing in payload on OCP which we don't want. We would want to hard fail on OCP.

Reply:
@sinnykumari There is a reference to the extensions image in the `image-references` file in OCP, so payloads won't be buildable without the image present. Also SCOS will at some point enable the extensions image, then this logic would have to be change again here. Does this address your concern or should I still change this?

Reply:
yeah, thinking more it sounds like I was being extra cautious which isn't required here.